rero_ils/modules/documents/api.py

```diff
@@ -388,10 +390,47 @@ class DocumentsIndexer(IlsRecordsIndexer):
 
     record_cls = Document
 
+    @classmethod
+    def _es_document(cls, record):
+        """Get the document from the corresponding index.
+
+        :param record: an item object
+        :returns: the elasticsearch document or {}
+        """
+        try:
+            es_item = current_search_client.get(
+                DocumentsSearch.Meta.index, record.id)
+            return es_item['_source']
+        except NotFoundError:
+            return {}
+
     def index(self, record):
         """Index an document."""
+        # get previous indexed version
+        es_document = self._es_document(record)
+
+        # call the parent index method
         return_value = super().index(record)
+
+        # index contributions
+        # TODO: reindex contributions only if it has been touched
         record.index_contributions(bulk=True)
+
+        # index document of the host document only if the title
+        # has been changed
+        # the comparison should be done on the dumps as _text is
+        # added for indexing
+        if not es_document \
+           or (record.dumps().get('title') != es_document.get('title')):
+            search = DocumentsSearch().filter(
+                'term', partOf__document__pid=record.pid)
+            ids = [doc.meta.id for doc in search.source().scan()]
+            # TODO: call a bulk indexing, but the bulk should also includes
+            # the partOf processing
+            for _id in ids:
+                doc = Document.get_record_by_id(_id)
+                doc.reindex()
```

Comment:
I'm a little bit confused about this. We begin to compare computed values to know if we need to reindex parents.
Dumping a document isn't a trivial operation because we load some linked resources from DB/ES (holdings,items, ...) In this case, we load all theses resource and don't use them.

I think we could just use the `title_format_text_head` function (https://github.com/rero/rero-ils/blob/1c8455b900c469fa90245575777a1679090e508c/rero_ils/modules/documents/utils.py#L225)) to compare  the title.

For my deep and personal opinion, I even think we need to reindex parent document if `record.get('title') != previous_record.get('title')` doesn't matters if a subfield of the subtitle change.